distribution/tools/server-cli/src/test/java/org/elasticsearch/server/cli/ServerCliTests.java

```diff
@@ -315,6 +320,15 @@ public void testServerExitsNonZero() throws Exception {
         assertThat(exitCode, equalTo(140));
     }
 
+    public void testServerCliCanLoadDifferentSecureSettingsImplementation() throws Exception {
+        // TODO: Change this to be using the Environment when we have an actual different implementation of SecureSettings
+        Command command = spy(newCommand());
+        doAnswer(args -> new MockSecureSettingsLoader()).when((ServerCli) command).secureSettingsLoader(any());
```

Comment:
This test class already contains a local ServerCli class specifically for doing this kind of mocking. Can we please stick with using that instead of adding mockito?

To following the existing pattern, you could either have a member var with an initialized loader, or just a boolean for whether it should be mocked. Then override the secureSettingsLoader method in the anonymous ServerCli class, and use the test member var to determine whether to return/construct a mock loader, or call super.